server/datastore/mysql/hosts_test.go

```diff
@@ -3320,10 +3320,17 @@ func testHostMDMAndMunki(t *testing.T, ds *Datastore) {
 	require.True(t, fleet.IsNotFound(err), err)
 
 	require.NoError(t, ds.SetOrUpdateMDMData(context.Background(), 432, true, "url", false))
+
+	enrolled, serverURL, installedFromDep, err := ds.GetMDM(context.Background(), 432)
+	require.NoError(t, err)
+	assert.True(t, enrolled)
+	assert.Equal(t, "url", serverURL)
+	assert.False(t, installedFromDep)
```

Comment:
Shouldn't we test `directIngestMDM` with `enrolled=false, missing or empty installed_from_dep, etc.`